src/Query/Builder.php

```diff
@@ -273,7 +273,11 @@ public function toMql(): array
                     $aggregations = blank($this->aggregate['columns']) ? [] : $this->aggregate['columns'];
 
                     if (in_array('*', $aggregations) && $function == 'count') {
-                        return ['count' => [$wheres, []]];
+                        if ($wheres) {
+                            return ['countDocuments' => [$wheres, []]];
+                        }
+
+                        return ['estimatedDocumentCount' => [[], []]];
```

Comment:
`estimatedDocumentCount` is syntactic sugar for executing the `count` command without a `filter` option. That allows it to return an estimated result by consulting the collection metadata instead of executing a query.

I'd be hesitant to silently opt users into this behavior through the same query builder syntax. Another consideration is that you cannot run `count` within a transaction at all (see: [Transactions: Count Operation](https://www.mongodb.com/docs/manual/core/transactions/#count-operation)).

IMO we should consistently use `countDocuments` here.

If we want to add support for `estimatedDocumentCount` we should invent some alternative syntax for that down the line.

This requires more overhead on the server side, but I think it'd ultimately serve users better since the counts will always be accurate and execute both within an without a transaction. I expect some users may also rely on the query builder's `count()` method for things like pagination (despite all the arguments against limit/skip-based pagination).

Reply:
You hit the bull's eye, options where not passed to `MongoDB\Collection::countDocuments()`.

Reply:
> options where not passed to MongoDB\Collection::countDocuments().

I'm not sure if it's a bug dating back to the original transaction implementation, but that seems like something worth tracking in a separate JIRA ticket for visibility.

Reply:
The options where already missing here: https://github.com/jenssegers/laravel-mongodb/blob/v3.9.5/src/Query/Builder.php#L287